Mathlib/Tactic/Simpa.lean

```diff
@@ -0,0 +1,45 @@
+/-
+Copyright (c) 2018 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Arthur Paulino, Mario Carneiro
+-/
+import Mathlib.Tactic.Core
+
+namespace Mathlib.Tactic
+
+open  Lean.Parser.Tactic Lean.Elab.Tactic
+
+-- move these?
+syntax simpArgs' := ((" only ")? " [" simpArg,+ "] ")?
+syntax withStx   := (" with " (colGt ident)+)?
+syntax usingStx  := (" using " term)?
+
+elab (name := simpa) "simpa" cfg?:(config)? disch?:(discharger)?
+    args?:simpArgs' wth?:withStx using?:usingStx : tactic => do
+  let cfg := cfg?.getOptional?
+  let args := args?.getOptional?
+  let nGoals := (← getUnsolvedGoals).length
+  try evalTactic (← `(tactic|simp $(cfg)? $(disch?.getOptional?)? $(args)?))
+  catch | _ => throwError "couldn't simplify the goal"
+  if (← getUnsolvedGoals).length < nGoals then
+    throwError "try 'simp' instead of 'simpa'"
+  match wth? with
+  | `(withStx| $[with $wth*]?) =>
+    match wth with
+    | none => pure ()
+    | some wth =>
+      for w in wth do
+        try evalTactic (← `(tactic|intro $w))
```

Comment:
This is not at all what the `with` flag is supposed to do.  The `with` flag specifies additional simp attributes that should be used; and should be directly passed to simp.  Unfortunately the simp side is not implemented yet, so for now we souold just ignore it.